recipes/dbg-macro/all/conanfile.py

```diff
@@ -13,31 +15,25 @@ class DbgMacroConan(ConanFile):
     settings = ("compiler", )
     no_copy_source = True
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = self.name + "-" + self.version
-        os.rename(extracted_dir, self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    def configure(self):
-        minimal_cpp_standard = "11"
-        if self.settings.get_safe("compiler.cppstd"):
-            tools.check_min_cppstd(self, minimal_cpp_standard)
+    def validate(self):
+        check_min_cppstd(self, 17)
 
-        if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "5":
+    def configure(self):
+        if self.settings.compiler == "gcc" and int(f"{self.settings.compiler.version}") < 5:
             raise ConanInvalidConfiguration(
-                "dbg-mcro can't be used by {0} {1}".format(
+                "dbg-macro can't be used by {0} {1}".format(
                     self.settings.compiler,
                     self.settings.compiler.version
                 )
             )
```

Comment:
This check should also go into the `validate()` method :)

Reply:
Thanks @RubenRBS for tip. Done.